Tweaks IV drip detaching and two sprites; Hot Foods Display now accepts utensils

Miraviel opened the pull request:
:cl:
rscadd: Added the "Detach IV Drip" verb under the Object tab.
tweak: Left-clicking an attached drip will now first detach the patient and only then it will remove the beaker. Drag and drop still works!
tweak: The Hot Foods Display machine now accepts and dispenses utensils.
imageadd: IV drips now have a slightly bigger container sprite and a wider warning light, for better visibility.
imageadd: Advanced first-aid kits are now purple in hands too, as opposed to being red.
/:cl:

![ivdripgif](https://i.imgur.com/QcaUO5j.gif)

1. For better accessibility. It checks if you are one tile away from both the attached person and the drip. 
2. I saw a few medical players not bothering with the drag-and-drop for detaching a patient, this should alleviate it. (If it's an unwelcome change, we should revert it asap.) It also feels more intuitive but it might only be me.
3. Cluttering the counter with utensils seemed to be lame and asking for a spoon in a mess hall, every time was... Weird too. Let adults decide whether they need a fork or not!
4. At least for me, I could never properly guess how much blood an IV drip had left. Hopefully, this tiny change would help others too without being an eyesore (see image above).
5. Purple kits used the old red kit in-hand sprite.

Question: Can we show the new drip_detach verb only when the IV drip is actually attached?

Miraviel committed:
Tweaks IV drip, sprites, Hot Food Display

Tweaks IV drip detaching, its sprite, and the advanced first-aid kit sprite; Hot Foods Display now accepts utensils

Miraviel committed:
Map fix

Something broke, whoops

Miraviel commented on code/game/objects/structures/iv_drip.dm:
```diff
@@ -134,6 +130,26 @@
+		to_chat(usr, "<span class='warning'>You are too far away from the [attached]!</span>")
+		return
+		
+	if(!usr.skill_check(SKILL_MEDICAL, SKILL_BASIC))
```
@afterthought2 Ginja asked on Discord why `STOP_PROCESSING(SSobj,src)` is only called when the needle is ripped out and why not when the skill check succeeds too. I only moved this entire process to a function, so I'm not quite sure about it.
(I remember you writing this part, if it wasn't you, sorry for the wrong ping!)

Hubblenaut replied:
Correct, it should stop processing regardless of whether the needle was ripped out or removed cleanly.

afterthought2 replied:
Probably my fault, yeah.

MistakeNot4892 replied:
As discussed above, the processing change should really be moved to the bottom of this block and deindented.

Miraviel replied:
Moved it and reindented.

Hubblenaut commented on code/game/objects/structures/iv_drip.dm:
```diff
@@ -53,14 +53,8 @@
+		drip_detach()
 	else if(ishuman(over_object))
 		hook_up(over_object, usr)
 	update_icon()
```
The icons will be updated twice now if `MouseDrop()` is used and `usr` has the required medical skills. Can remove it from here and put it at the end of `drip_detach()` so that it is only called once in every case.

afterthought2 replied:
Could use the new update icon subsystem too. To do that, replace `update_icon` with `queue_icon_update`.

Miraviel replied:
Replaced every `update_icon()` with `queue_icon_update()`. @afterthought2 
Removed `queue_icon_update()` from `rip_out()` and from `MouseDrop()` and moved it to the end of `drip_detach()`. @Hubblenaut

Hubblenaut commented on code/game/objects/structures/iv_drip.dm:
```diff
@@ -134,6 +130,26 @@
+		rip_out()
+		STOP_PROCESSING(SSobj,src)
+	else
+		visible_message("\The [attached] is taken off \the [src]")
```
I know it was already missing before, but it'd be nice to have punctuation at the end of this message.

Miraviel replied:
Tactical dot placed.

MistakeNot4892 commented on code/game/machinery/kitchen/smartfridge.dm:
```diff
@@ -132,7 +132,7 @@
 
 /obj/machinery/smartfridge/foods/accept_check(var/obj/item/O as obj)
-	if(istype(O,/obj/item/weapon/reagent_containers/food/snacks))
+	if(istype(O,/obj/item/weapon/reagent_containers/food/snacks) || istype(O,/obj/item/weapon/material/kitchen/utensil/))
```
This is going to sound petty, but the trailing `/` on this path should be removed.

Miraviel replied:
Whoops, I have no idea why I put it there - removed it.

Miraviel committed:
Update iv_drip.dm

Replaced icon_update() with queue_icon_update()
Moved STOP_PROCESSING(SSobj,src) to the end of the block, reindented.

Miraviel committed:
Update smartfridge.dm

Miraviel committed:
Update iv_drip.dm

Miraviel commented:
Addressed every issue, however, I cannot test it locally until this evening (GMT).

Miraviel committed:
Update iv_drip.dm

MistakeNot4892 commented:
Forgot to note that this could really use a squash.
